practice/validator/site.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+/**
+ * The Validator takes in the user's input as they type into any given text box
+ * in the form and uses kestroke events to visually que the user that their
+ * input is either valid or not, by filling the field with a yellow background
+ * while invalid.
+ */
+
+/**
+ * Adds the invalid class to the given field (distinguished by a class).
+ *
+ * The invalid class has a yellow background in the site.css file.
+ */
+function makeFieldYellow(inputclass) {
+  $('input.' + inputclass).addClass('invalid');
+}
+
+/**
+ * Removes the invalid class from a given field (distinguished by a class).
+ */
+function removeYellowFromField(inputclass) {
+  $('input.' + inputclass).removeClass('invalid');
+}
+
+/**
+ * Takes in a field (distinguished by a class) and the entry that has been
+ * typed into the textbox of the given field, and returns a boolen representing
+ * whether or not the entry is valid for the given field.
+ */
+function checkIfValid(inputclass, entry) {
+  if (inputclass === 'fullname') {
+    return /(\w+)\s(\w+)/.test(entry);
```

Comment:
\w is defined as: Matches any letter, number or underscore. Which means that this would accept my name as Stephen Muller, or Stephen Mull3r, or s_ 4 even. What you really want is something that matches any a-z character which would look like [a-z] (only lower case) or [a-zA-Z] (not case sensitive). The '+' is absolutely the right call.